### ISSUE TITLE
Handle transaction rollbacks

### ISSUE THREAD

**Issue by lukebryant** [romgar/django-dirtyfields#148]
Currently, if we do:

```
tm.field = 'new field value'
tm.get_dirty_fields() #returns {'field': 'original field value'}
try:
    with transaction.atomic(): 
        tm.save()
        raise Exception
except Exception:
     pass
tm.get_dirty_fields() #returns {}
```

This behaviour is problematic, because field in-memory and database values are still different.

I've thought about this, and I don't see any way of solving it. 

We could have a property `uncommitted_dirty_fields`, which we could clear in an `on_commit()` signal, but I can't think of a way of setting `dirty_fields = uncommitted_dirty_fields` in the case of a rollback, since there is no `on_rollback()` signal.

**Comment by romgar** [comment]
Hey @lukebryant!
Interesting problem...
We could create a custom decorator / context manager that would wrap the transaction atomic and add this missing logic, but that's not really ideal as people would have to know that they should use this custom feature from `dirtyfields` instead of the default Django API...

**Comment by lukebryant** [comment]
> Hey @lukebryant!
> Interesting problem...
> We could create a custom decorator / context manager that would wrap the transaction atomic and add this missing logic, but that's not really ideal as people would have to know that they should use this custom feature from `dirtyfields` instead of the default Django API...

I was thinking this wouldn't work because the context manager would have no way of getting references to all the models that are changed within its scope. Am I missing something, or did you mean that the context manager would take a list of potentially mutating models as arguments?

**Comment by romgar** [comment]
Yeah you would need to pass these as arguments. Far from ideal for sure...

**Comment by romgar** [comment]
Right now, I would at least document the issue and ask people to manually save and restore dirty fields for their affected model in case of rollback.
